python/mxnet/metric.py

```diff
@@ -423,8 +501,11 @@ def update(self, labels, preds):
 
             check_label_shapes(label, pred_label)
 
-            self.sum_metric += (pred_label == label).sum()
+            num_correct = (pred_label == label).sum()
+            self.sum_metric += num_correct
+            self.global_sum_metric += num_correct
```

Comment:
They will be different when you call `reset_local` (which is called from `Speedometer` for example) - this will reset the local versions of metrics while keeping the global versions intact (which was the point of this PR - to enable computing both per-batch and per-epoch statistics using only single computation).

Reply:
That said, this comment made me think about this again and I found a bug in how I handle global statistics in `F1` and `MCC` metrics - fix and tests incoming. Thank you!